### ISSUE TITLE
equals rule processing change with 0.18?

### ISSUE THREAD

**Issue by sconnard** [aide/aide#154]
With AIDE 0.18, AIDE seems to totally ignore any directories lower than an equals rule directory.  Prior to this you could create secondary rules that allowed tracking of files below an equals directory.

=/test/dir1 R
=/test/dir1/dir1a R
/test/dir2 R

Suppose I want to track dir1 and specific things under dir1 but not everything. I create an =dir1 rule to just track the directory, then I create a =dir1/dir1a to track a sub-directory.

Under AIDE 0.18.2 dir1a is not in the database but in earlier versions of AIDE it is.

**Comment by hvhaugwitz** [comment]
Thanks for your bug report.

Unfortunately I cannot reproduce your issue with latest stable release AIDE `0.18.2`:

```
$ aide --config <(printf "=/test/dir1 R\n=/test/dir1/dir1a R\n/test/dir2 R\n") --path-check d:/test/dir1
[X] d '/test/dir1': equal rule: '=/test/dir1 (none) l+p+u+g+s+c+m+i+n+md5+ftype' (/dev/fd/63:1: '=/test/dir1 R')
$ aide --config <(printf "=/test/dir1 R\n=/test/dir1/dir1a R\n/test/dir2 R\n") --path-check d:/test/dir1/dir1a
[X] d '/test/dir1/dir1a': equal rule: '=/test/dir1/dir1a (none) l+p+u+g+s+c+m+i+n+md5+ftype' (/dev/fd/63:2: '=/test/dir1/dir1a R')
$ aide --version
AIDE 0.18.2
[...]
```

Please provide more information about how to reproduce your problem, preferably the output of a `--path-check` AIDE run for the misbehaving path using `rule` log level.

**Comment by hvhaugwitz** [comment]
@sconnard Can you please provide more information about this issue as requested in my previous message?

**Comment by hvhaugwitz** [comment]
This issue was raised also on the AIDE user mailing list (https://www.ipi.fi/pipermail/aide/2023-June/001766.html).

**Comment by sconnard** [comment]
Hannes,

I have tested the fix in Aide 18.5 and I’m very happy with it.

Thank you!
